web/client/components/map/openlayers/plugins/WMTSLayer.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2015, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+var Layers = require('../../../../utils/openlayers/Layers');
+var ol = require('openlayers');
+var objectAssign = require('object-assign');
+const CoordinatesUtils = require('../../../../utils/CoordinatesUtils');
+const {isArray} = require('lodash');
+const SecurityUtils = require('../../../../utils/SecurityUtils');
+const mapUtils = require('../../../../utils/MapUtils');
+
+function wmtsToOpenlayersOptions(options) {
+    // NOTE: can we use opacity to manage visibility?
+    return objectAssign({}, options.baseParams, {
+        LAYERS: options.name,
+        STYLES: options.style || "",
+        FORMAT: options.format || 'image/png',
+        TRANSPARENT: options.transparent !== undefined ? options.transparent : true,
+        SRS: CoordinatesUtils.normalizeSRS(options.srs || 'EPSG:3857', options.allowedSRS),
+        CRS: CoordinatesUtils.normalizeSRS(options.srs || 'EPSG:3857', options.allowedSRS),
+        TILED: options.tiled || false,
+        VERSION: options.version || "1.3.0"
+    }, options.params || {});
+}
+
+function getWMSURLs( urls ) {
+    return urls.map((url) => url.split("\?")[0]);
+}
+
+Layers.registerType('wmts', {
+    create: (options) => {
+        const urls = getWMSURLs(isArray(options.url) ? options.url : [options.url]);
+        const queryParameters = wmtsToOpenlayersOptions(options) || {};
+        urls.forEach(url => SecurityUtils.addAuthenticationParameter(url, queryParameters));
+        let matrixIds = new Array(22);
```

Comment:
Tile matrix zoom levels are not 22 everywhere. Even when auto-generation is used, the list of resolutions and matrixIds can be variable. In general we should follow the information from WMTS. 
Keep in mind these 2 examples: 
 * http://basemap.nationalmap.gov/arcgis/rest/services/USGSHydroNHD/MapServer/WMTS/1.0.0/WMTSCapabilities.xml
 * http://basemap.nationalmap.gov/arcgis/rest/services/USGSHydroNHD/MapServer/WMTS/1.0.0/WMTSCapabilities.xml

As you see the identifier and the list are not the same.